src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -79,4 +97,26 @@ public function elementValidate($element, FormStateInterface $form_state, $form)
 
     // todo: Check the writing permission for the current user.
   }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getWidgetState(array $parents, $field_name, FormStateInterface $form_state) {
+    $widgetState = parent::getWidgetState($parents, $field_name, $form_state);
+
+    if (empty(self::$info)) {
+      return $widgetState;
+    }
+    $info = self::$info;
+    $results = \Drupal::entityQuery('og_membership')
```

Comment:
Also, doesn't the work I started to do on loading make this less of a problem? The widget should get any values from the field item (as it does now) and use that. Rather than any funky stuff in the widget itself?